ImageGenerator/include/painter.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "image.h"
+
+#include <array>
+#include <vector>
+
+namespace IMAGE {
+
+struct Coord {
+    double x, y;
+};
+
+struct Color4 {
+    double r, g, b;
+    double a;
+};
+
+struct Color3 {
+    double r, g, b;
+};
+
+using Color = Color4;
+
+struct Circle {
+    Coord center;
+    double radius;
+};
+
+struct Diamond {
+    Coord center;
+    double radius;
+};
+
+struct Rectangle {
+    Coord p1, p2;
+};
+
+struct Triangle {
+    Coord p1, p2, p3;
+};
+
+struct Line {
+    Coord p1, p2;
+    double radius;
+};
+
+struct Painter {
+    void Draw(Image& image, const Circle& circle, const Color& color) const;
+    void Draw(Image& image, const Diamond& diamond, const Color& color) const;
+    void Draw(Image& image, const Rectangle& rectangle, const Color& color) const;
```

Comment:
Draw(Image& image, const Image& sourceImage, const Rectangle& rectangle);

Consider :  deside order of input. (rect+imge vs image+rect)